protocol/src/main/java/org/fidoalliance/fdo/protocol/serialization/ManufacturingInfoDeserializer.java

```diff
@@ -38,8 +56,23 @@ public ManufacturingInfo deserialize(JsonParser jp, DeserializationContext ctxt)
 
     info.setKeyType(PublicKeyType.fromNumber(node.get(index++).intValue()));
     info.setKeyEnc(PublicKeyEncoding.fromNumber(node.get(index++).intValue()));
-    info.setSerialNumber(node.get(index++).textValue());
-    info.setDeviceInfo(node.get(index++).textValue());
+
+    String serialNumber = node.get(index++).textValue();
+
+    if (isValidString(serialNumber)) {
+      info.setSerialNumber(serialNumber);
+    } else {
+      throw new InvalidMessageException("Invalid Serial Number");
```

Comment:
Make changes to ensure that CI is passing.

Reply:
Added javadoc for the funciton